Unsupported operand type(s) for +: 'NoneType' and 'int' in genome matcher module

An error reported from a colleague from the MatchBox node (Broad institute). Needs fixing ASAP!

```
###### Running genome matcher module ######
2021-01-05 17:33:56 clinical-db.scilifelab.se patientMatcher.match.genotype_matcher[22370] INFO Query patient has 1 genotype features.
2021-01-05 17:33:56 clinical-db.scilifelab.se patientMatcher.match.genotype_matcher[22370] INFO Each GT feature will contribute with a weight of 0.75 to a total GT score (max GT score is 0.75)
[2021-01-05 17:33:56 +0100] [22370] [ERROR] Error handling request /match
Traceback (most recent call last):
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 134, in handle
    self.handle_request(listener, req, client, addr)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 175, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/werkzeug/middleware/proxy_fix.py", line 228, in __call__
    return self.app(environ, start_response)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask_negotiate.py", line 14, in wrapper
    return fn(*args, **kwargs)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/flask_negotiate.py", line 27, in wrapper
    return fn(*args, **kwargs)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/patientMatcher/server/views.py", line 288, in match_internal
    current_app.db, query_patient, max_pheno_score, max_geno_score, max_results, score_threshold
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/patientMatcher/match/handler.py", line 83, in internal_matcher
    geno_matches = genomatch(database, patient_obj["genomicFeatures"], max_geno_score)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/patientMatcher/match/genotype_matcher.py", line 49, in match
    variants = gtfeatures_to_variants(gt_features)
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/patientMatcher/parse/patient.py", line 235, in gtfeatures_to_variants
    lifted_variants = lift_variant(feature["variant"])
  File "/home/hiseq.clinical/miniconda/envs/P_pmatcher/lib/python3.6/site-packages/patientMatcher/parse/patient.py", line 196, in lift_variant
    variant.get("end") + 1,
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
127.0.0.1 - - [05/Jan/2021:17:33:56 +0100] "POST /match HTTP/1.0" 500 0 "-" "-"
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 ### Added
 ### Changed
 ### Fixed
+- Liftover crashing when no end coordinate is not provided
 
 ## [2.4] - 2020-12-30
 ### Added

--- a/patientMatcher/parse/patient.py
+++ b/patientMatcher/parse/patient.py
@@ -193,7 +193,7 @@ def lift_variant(variant):
         variant.get("assembly"),
         variant.get("referenceName"),
         variant.get("start") + 1,  # coordinates are 0-based in MatchMaker
-        variant.get("end") + 1,
+        variant.get("end") + 1 if variant.get("end") else variant.get("end"),
     )
 
     if mappings is None:
@@ -207,6 +207,8 @@ def lift_variant(variant):
         lifted["assembly"] = mapped["assembly"]
         lifted["referenceName"] = mapped["seq_region_name"]
         lifted["start"] = mapped["start"] - 1  # conver back to 0-based coordinates
+        if variant.get("end") is None:
+            continue
         lifted["end"] = mapped["end"] - 1
 
         lifted_vars.append(lifted)

--- a/patientMatcher/utils/variant.py
+++ b/patientMatcher/utils/variant.py
@@ -2,14 +2,14 @@
 import patientMatcher.utils.ensembl_rest_client as ensembl_client
 
 
-def liftover(build, chrom, start, end):
+def liftover(build, chrom, start, end=None):
     """Perform variant liftover using Ensembl REST API
 
     Accepts:
         build(str): genome build: GRCh37 or GRCh38
         chrom(str): 1-22,X,Y,MT
         start(int): start coordinate
-        stop(int): stop coordinate
+        stop(int): stop coordinate or None
 
     Returns
         mappings(list of dict):
@@ -25,7 +25,7 @@ def liftover(build, chrom, start, end):
             client.server,
             "map/human",
             build,
-            f"{chrom}:{start}..{end}",
+            f"{chrom}:{start}..{end or start}",  # End variant provided is not required
             f"{assembly2}?content-type=application/json",
         ]
     )

